tests/geometry/test_intersections.py

```diff
@@ -1,9 +1,2539 @@
+        p_1 = np.ones(3)
+        p_2 = 2 * p_1
+        p_3 = 0 * p_1
+        p_4 = 3 * p_1
+
+        p_int_1 = intersections.segments_3d(p_1, p_2, p_3, p_4)
+        p_int_2 = intersections.segments_3d(p_2, p_1, p_3, p_4)
+        p_int_3 = intersections.segments_3d(p_1, p_2, p_4, p_3)
+        p_int_4 = intersections.segments_3d(p_2, p_1, p_4, p_3)
+
+        p_known_1 = p_1.reshape((-1, 1))
+        p_known_2 = p_2.reshape((-1, 1))
+
+        l1 = np.array(
+            [p_int_1, p_int_1, p_int_2, p_int_2, p_int_3, p_int_3, p_int_4, p_int_4]
+        )
+        l2 = np.array(
+            [
+                p_known_1,
+                p_known_2,
+                p_known_1,
+                p_known_2,
+                p_known_1,
+                p_known_2,
+                p_known_1,
+                p_known_2,
+            ]
+        )
+
+        for p_int, p_known in zip(l1, l2):
+            assert np.min(np.sum(np.abs(p_int - p_known), axis=0)) < 1e-8
+
+    @pytest.fixture
+    def assert_equal_segments_3d(self):
+        def _assert_equal_segments_3d(p_1, p_2, p_3, p_4):
+            p_int = intersections.segments_3d(p_1, p_2, p_3, p_4)
+            p_known_1 = p_1.reshape((-1, 1))
+            p_known_2 = p_4.reshape((-1, 1))
+            assert np.min(np.sum(np.abs(p_int - p_known_1), axis=0)) < 1e-8
+            assert np.min(np.sum(np.abs(p_int - p_known_2), axis=0)) < 1e-8
+
+        return _assert_equal_segments_3d
+
+    def test_segments_partly_overlap(self, assert_equal_segments_3d):
+        p_1 = np.ones(3)
+        p_2 = 3 * p_1
+        p_3 = 0 * p_1
+        p_4 = 2 * p_1
+
+        assert_equal_segments_3d(p_1, p_2, p_3, p_4)
+
+    def test_random_incline(self, assert_equal_segments_3d):
+        p_1 = np.random.rand(3)
+        p_2 = 3 * p_1
+        p_3 = 0 * p_1
+        p_4 = 2 * p_1
+
+        assert_equal_segments_3d(p_1, p_2, p_3, p_4)
+
+    def test_segments_aligned_with_axis(self, assert_equal_segments_3d):
+        p_1 = np.array([0, 1, 1])
+        p_2 = 3 * p_1
+        p_3 = 0 * p_1
+        p_4 = 2 * p_1
+
+        assert_equal_segments_3d(p_1, p_2, p_3, p_4)
+
+    def test_constant_y_axis(self):
+        p_1 = np.array([1, 0, -1])
+        p_2 = np.array([1, 0, 1])
+        p_3 = np.array([1.5, 0, 0])
+        p_4 = np.array([0, 0, 1.5])
+
+        p_int = intersections.segments_3d(p_1, p_2, p_3, p_4)
+        p_known = np.array([1, 0, 0.5]).reshape((-1, 1))
+        assert np.min(np.sum(np.abs(p_int - p_known), axis=0)) < 1e-8
+
+
+class TestSweepAndPrune:
+    def pairs_contain(self, pairs, a):
+        for pi in range(pairs.shape[1]):
+            if a[0] == pairs[0, pi] and a[1] == pairs[1, pi]:
+                return True
+        return False
+
+    def test_no_intersection(self):
+        x_min = np.array([0, 2])
+        x_max = np.array([1, 3])
+
+        pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+        assert pairs.size == 0
+
+    @pytest.fixture
```

Comment:
Is there any reason to make this a fixture? Isn't it really only a helper function?

Reply:
I guess that might be true. Should I change it to be a function or just keep it as is?

Reply:
Please change it.